src/modules/app/routes/categoriesRedirect.js

```diff
@@ -28,11 +29,13 @@ export const categoriesRedirectRoute = {
       // todo error handling
     }
 
-    try {
-      const category = categoriesPayload.data.getCategoryById(Number(categoryId))
-      dispatch(redirect(goToCategories(city, language, category.path)))
-    } catch (e) {
-      dispatch(redirect(goToNotFound(city, language)))
+    if (categoriesPayload.data instanceof CategoriesMapModel) {
```

Comment:
Why do you need to check that its of the right "type"?
And what should happen if it wasn't the right type?

Reply:
It is required by flow, otherwise we get an error here because we can't be sure that the payload data is a CategoriesMapModel. Will do this in WEBAPP-203, along with most of the other error handling stuff.

Reply:
actually I don't think we need any error handling here at all. The data of the payload must not be something different than the corresponding model (or an array of it) and imo we don't have to handle the case that it is not the right data anyway, we should just not fetch or redirect than.

Reply:
on the other hand if the data is null, the payload has an error, which will be handled in the Switcher component (or a FailureComponent rendered in Switcher). Will do this in WEBAPP-203